system/lib/libc/emscripten_time.c

```diff
@@ -32,7 +32,16 @@ double emscripten_get_now_res();
 
 __attribute__((__weak__))
 void tzset() {
-  _tzset_js(&timezone, &daylight, tzname);
+  static pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+  static _Atomic bool done_init = false;
+  if (!done_init) {
+    pthread_mutex_lock(&lock);
```

Comment:
Do we need manual locking? I believe function-local statics will be guarded by proper locks by the compiler.

Reply:
Does that work in C though?  (I seems to remember `static int foo = callFunc();` is a c++ thing)

Reply:
Ah, it could be C++ only... searching for this now, all the links are in C++ and some even say it is only in recent C++ versions.

Reply:
Confirmed this doesn't work in C: `emscripten_time.c:37:19: error: initializer element is not a compile-time constant
  static bool x = _tzset_js(&timezone, &daylight, tzname);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
`

Reply:
Interesting, thanks for checking.

This actually seems like a reason to write such code in C++, or a subset of C++ that is almost C but has this benefit... but just a thought.

Reply:
Maybe.. if we can be sure we don't accidentally pull in C++ library code then maybe it would be a good idea. On the other hand I don't think this is a common pattern yet.